src/file_system/mod.rs

```diff
@@ -473,6 +494,64 @@ impl Directory {
         Directory::mkdir(Label::root(), Repo::repo_directory())
     }
 
+    /// Get the total size, in bytes, of a `Directory`. The size is
+    /// the sum of all files that can be reached from this `Directory`.
```

Comment:
This implies recursively getting the size for all sub-directories, is that true? How expensive is this operation? Thinking esp. about not so modern storage devices and/or congested user machines.